### ISSUE TITLE
rtcwake fails to suspend (wakes instantly) from commandline

### ISSUE THREAD

**Issue by w-barath** [karelzak/util-linux#527]
With a wireless USB keyboard, issue the command:

`$ sudo rtcwake -m mem -s 60`

result (cut from dmesg):

> [2491353.279700] smpboot: CPU 3 is now offline
> [2491353.299241] ACPI: Low-level resume complete
> [2491353.299293] ACPI : EC: EC started
> [2491353.299293] PM: Restoring platform NVS memory
> [2491353.299612] Suspended for 0.116 seconds

What's going on here?  It's simple: wireless USB keyboard continues to send events for some fraction of a second after the return key is pressed, resulting in instant wake-up

Solution: if stdin is a TTY, usleep(1000) before invoking sleep via sysfs.

This issue can also cause failure to suspend via a mouse click as well, so maybe it would be best to add a usleep(300) at a minimum regardless of whether it's a TTY.

Or I guess you could monitor /dev/input/* until a certain amount of time without activity... way way more complicated.

If these solutions seem unacceptable to the maintainer, then can I please recommend a note in the man page about users ensuring that all resume input devices are idle for a reasonable amount of time before running the command.  That will probably result in some more feedback from other users asking how they're meant to ensure the input devices are idle when they are pressing enter to run the command, lol.

**Comment by karelzak** [comment]
Do you want to say that (for example) "sleep 5; sudo rtcwake -m mem -s 60" fixes the problem for you?

Frankly, sleep() in tools is often poor design indicator. The ideal solution would be to wait in kernel for input inactivities. I'll think about it.

Add a hint to the man page is definitely good idea. The issue is no only wireless USB, but also things like fstrim, etc.

**Comment by kerolasa** [comment]
I just added commit c9658d6fb2faf7a756c8aa5242bf9f3661a37301 to my git, see above link. That commit will add similar sort of settle down wait as what is used in script(1) when waiting child shell. Notice that I am not sure if discarding input is always sane thing, by what I mean this might not be a change we need or want.

**Comment by karelzak** [comment]
I have applied the patch from @kerolasa . It seems good enough to wait by poll(). In the worst case (stdin still active) it will wait for 2s before sleep.
